### DISCUSSION TITLE
一对多可以支持只有一个子实体的，非集合导航属性么？

### DISCUSSION THREAD

**Discussion by lif-enable** [dotnetcore/FreeSql#1082] [Q&A] [answered]
```
class A{
int Id{get;set;}
 B b{get;set;} 
}
class B{
int Id{get;set;}
 A A{get;set;}
int AId{get;set;}
}
```
类似这种关系，A.b (null 或只有一个)  , freesql 能不能将其看作一对多模式 ，使用级联保存。
目前会报错，类型的属性  不是 OneToMany 或 ManyToMany 特性

**Comment by 2881099** [comment, marked as answer]
非集合属性，手工保存也很方便，所以没有做自动化。

因为多对一，自动级联保存很危险，比如保存 Topic，不应该把 TopicType 也保存。

然后一对一，和多对一目前没标识区分，所以就是现在这样。

**Comment by lif-enable** [comment]
多謝了

**Comment by lif-enable** [comment]
有沒有类似efcore 里的owner 类型的模式
